.github/workflows/PR.yaml

```diff
@@ -45,7 +46,24 @@ jobs:
               - '**.rs'
               - '.github/**'
               - '**.toml'
-
+      - id: timestamp
+        run: |
+          export timestamp=$(date +%s)
+          echo "timestamp=$timestamp" >> $GITHUB_OUTPUT
+          echo "::group::DEBUG"
+          echo "timestamp=$timestamp"
+          echo "::endgroup::"
+      - name: Get Current version
+        id: version
+        env:
+          timestamp: ${{ steps.timestamp.outputs.timestamp }}
+        run: |
+          version=$(grep "version =" libsovtoken/Cargo.toml | grep -oP "[\d\.]+\d(-(dev|rc)(\d+))?")
+          if echo $version | grep -oP "\-(dev|rc)[0-9]?" >> /dev/null ;then
+            echo "newVersion=$(echo $version | sed -rE 's~(([0-9]\.)+[0-9]-(dev|rc))([0-9]+)~\1$timestamp~')" >> $GITHUB_OUTPUT
+          else
+            echo "newVersion=$version-dev$timestamp" >> $GITHUB_OUTPUT
+          fi
```

Comment:
Recommend moving this to a action as it's happening on both PR and Push workflows.